tools/build_everything.py

```diff
@@ -146,8 +146,12 @@
         base_source_paths = options.source_dir
         if not base_source_paths:
             base_source_paths = ['.']
-        
-        all_tests = find_tests(base_source_paths[0])
+       
+        all_tests = {}
+        for platform in platforms:
+            for toolchain in toolchains:
+                platform_toolchain_tests = find_tests(base_source_paths[0], platform, toolchain)
+                all_tests.update(platform_toolchain_tests)
```

Comment:
Shouldn't `find_tests()` care of this? I just recently started looking at the build system and I'm not that familiar with general architecture but this seems to me good idea regarding separation of concern.